drivers/sx127x/sx127x_getset.c

```diff
@@ -0,0 +1,735 @@
+
+    sx127x_reg_write(dev, SX127X_REG_PARAMP, SX127X_RF_PARAMP_0050_US);
+
+    if ((pa_config & SX127X_RF_PACONFIG_PASELECT_PABOOST)
+        == SX127X_RF_PACONFIG_PASELECT_PABOOST) {
+        if (power > 17) {
+            pa_dac = ((pa_dac & SX127X_RF_PADAC_20DBM_MASK) |
+                      SX127X_RF_PADAC_20DBM_ON);
+        } else {
+            pa_dac = ((pa_dac & SX127X_RF_PADAC_20DBM_MASK) |
+                      SX127X_RF_PADAC_20DBM_OFF);
+        }
+        if ((pa_dac & SX127X_RF_PADAC_20DBM_ON) == SX127X_RF_PADAC_20DBM_ON) {
+            if (power < 5) {
+                power = 5;
+            }
+            if (power > 20) {
+                power = 20;
+            }
+
+            pa_config = ((pa_config & SX127X_RF_PACONFIG_OUTPUTPOWER_MASK) |
+                         (uint8_t)((uint16_t)(power - 5) & 0x0F));
+        } else {
+            if (power < 2) {
+                power = 2;
+            }
+            if (power > 17) {
+                power = 17;
+            }
+
+            pa_config = ((pa_config & SX127X_RF_PACONFIG_OUTPUTPOWER_MASK) |
+                         (uint8_t)((uint16_t)(power - 2) & 0x0F));
+        }
+    } else {
+        if (power < -1) {
+            power = -1;
+        }
+        if (power > 14) {
+            power = 14;
+        }
+
+        pa_config = ((pa_config & SX127X_RF_PACONFIG_OUTPUTPOWER_MASK) |
+                     (uint8_t)((uint16_t)(power + 1) & 0x0F));
+    }
+
+    sx127x_reg_write(dev, SX127X_REG_PACONFIG, pa_config);
+    sx127x_reg_write(dev, SX127X_REG_PADAC, pa_dac);
+}
+
+uint16_t sx127x_get_preamble_length(sx127x_t *dev)
+{
+    return dev->settings.lora.preamble_len;
+}
+
+void sx127x_set_preamble_length(sx127x_t *dev, uint16_t preamble)
+{
+    dev->settings.lora.preamble_len = preamble;
+
+    sx127x_reg_write(dev, SX127X_REG_LR_PREAMBLEMSB,
+                     (preamble >> 8) & 0xFF);
+    sx127x_reg_write(dev, SX127X_REG_LR_PREAMBLELSB,
+                     preamble & 0xFF);
+}
+
+void sx127x_set_rx_timeout(sx127x_t *dev, uint32_t timeout)
+{
+    dev->settings.lora.rx_timeout = timeout;
+}
+
+void sx127x_set_tx_timeout(sx127x_t *dev, uint32_t timeout)
+{
+    dev->settings.lora.tx_timeout = timeout;
+}
+
+void sx127x_set_symbol_timeout(sx127x_t *dev, uint16_t timeout)
+{
+    dev->settings.lora.rx_timeout = timeout;
+
+    uint8_t config2_reg = sx127x_reg_read(dev, SX127X_REG_LR_MODEMCONFIG2);
+    config2_reg &= SX127X_RF_LORA_MODEMCONFIG2_SYMBTIMEOUTMSB_MASK;
+    config2_reg |= (timeout >> 8) & ~SX127X_RF_LORA_MODEMCONFIG2_SYMBTIMEOUTMSB_MASK;
+    sx127x_reg_write(dev, SX127X_REG_LR_MODEMCONFIG2, config2_reg);
+    sx127x_reg_write(dev, SX127X_REG_LR_SYMBTIMEOUTLSB,timeout & 0xFF);
+}
+
+void sx127x_set_iq_invert(sx127x_t *dev, bool iq_invert)
+{
+    dev->settings.lora.iq_inverted = iq_invert;
+
+    uint8_t invertiq_reg = sx127x_reg_read(dev, SX127X_REG_LR_INVERTIQ);
+    invertiq_reg &= SX127X_RF_LORA_INVERTIQ_RX_MASK;
```

Comment:
Is it really usefull ? this bit is all alone in the register. I think we can optimize here.